rfi_file_monitor/utils/s3.py

```diff
@@ -1,14 +1,21 @@
+from __future__ import annotations
+
 import boto3.s3.transfer
 
 from ..file import File
+from . import ExitableThread
 
 import os
 import hashlib
 from typing import Optional, Union
+import threading
+import logging
+
+logger = logging.getLogger(__name__)
 
 KB = 1024
 MB = KB * KB
-TransferConfig = boto3.s3.transfer.TransferConfig(max_concurrency=1, multipart_chunksize=8*MB, multipart_threshold=8*MB)
+TransferConfig = boto3.s3.transfer.TransferConfig(max_concurrency=1, multipart_chunksize=8*MB, multipart_threshold=8*MB, use_threads=False)
```

Comment:
does this impact large files at all? Other than making them slower to upload

Reply:
Hmmm, let me quickly try that. I put this option in as it will be necessary for making the operations cancellable when that boto/s3transfer PR is merged in.